packages/experimental/Menu/src/MenuList/MenuList.tsx

```diff
@@ -33,18 +35,53 @@ export const MenuList = compose<MenuListType>({
   ...stylingSettings,
   slots: {
     root: MenuStack,
+    focusZone: FocusZone,
   },
   useRender: (userProps: MenuListProps, useSlots: UseSlots<MenuListType>) => {
     const menuList = useMenuList(userProps);
     const contextValue = useMenuListContextValue(menuList);
     const Slots = useSlots(menuList);
 
+    /**
+     * On macOS, focus isn't placed by default on the first focusable element. We get around this by focusing on the inner FocusZone
+     * hosting the menu. For whatever reason, to get the timing _just_ right to actually focus, we need an additional `setTimeout`
+     *  on top of the `useLayoutEffect` hook.
+     */
+    const focusZoneRef = React.useRef<IFocusable>(null);
+
+    React.useLayoutEffect(() => {
+      if (Platform.OS === 'macos') {
+        setTimeout(() => {
+          focusZoneRef.current?.focus();
+        }, 0);
+      }
+    });
+
     return (_final: MenuListProps, children: React.ReactNode) => {
-      return (
-        <MenuListProvider value={contextValue}>
-          <Slots.root>{children}</Slots.root>
-        </MenuListProvider>
-      );
+      if (Platform.OS === 'macos') {
+        return (
+          <MenuListProvider value={contextValue}>
+            <Slots.root>
+              <Slots.focusZone
```

Comment:
I think it would be cleaner/less duplicate to so something like

```tsx
const content = Platform.OS === 'macos' ? (
         <Slots.focusZone
                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
                // @ts-ignore
                enableFocusRing={false}
                componentRef={focusZoneRef}
                defaultTabbableElement={focusZoneRef}
                focusZoneDirection={'vertical'}
              >
                {children}
              </Slots.focusZone>
) : (
    {children}
)

return (
<MenuListProvider value={contextValue}>
            <Slots.root>
                {content}
            </Slots.root>
</MenuListProvider>
);
```

Less duplication of the render code which means less likely that things will be missed if we need to add things here.

Reply:
I think I tried that before and ran into weird typing errors. I'll try once more.

Reply:
This worked, thank you!